test/NuGet.Clients.Tests/NuGet.CommandLine.Test/ExtractOptionsTests.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace NuGet.CommandLine.Test
+{
+    public class ExtractOptionsTests
+    {
+
+        [Fact]
+        public void ExtractOption_PassOptionNoHttpCache_NoHttpCacheShouldBeSet()
+        {
+            //setup
+            RestoreCommand command = new RestoreCommand();
+            List<string> args = new() { "-NoHttpCache" };
+            ICommandManager manager = new CommandManager();
+            CommandLineParser commandLineParser = new CommandLineParser(manager);
+
+            //execute extract options
+            commandLineParser.ExtractOptions(command, args.GetEnumerator());
+
+            //assert
+            Assert.True(command.NoHttpCache);
+        }
+
+        [Fact]
+        public void ExtractOption_DoNotPassOptionNoHttpCache_NoHttpCacheShouldNotBeSet()
```

Comment:
Let's make it explicit in the test case name what we are expecting. Unless we are relying on the default behavior of the command class when no option for NoHttpCache is being passed (in which case this test would need to change), we should change `ShouldNotBeSet` to tell us what exactly that means, which is that we are expecting it to be set to false.

```suggestion
        public void ExtractOption_DoNotPassOptionNoHttpCache_NoHttpCacheShouldBeFalse()
```